dev/BulkLoader.php

```diff
@@ -143,7 +152,83 @@ public function load($filepath) {
 			DataObject::get($this->objectClass)->removeAll();
 		}
 
-		return $this->processAll($filepath);
+		$files = $this->splitFile($filepath);
+
+		$result = null;
+
+		foreach ($files as $file) {
+			$next = $this->processAll($file);
+
+			if ($result instanceof BulkLoader_Result) {
+				$result->merge($next);
+			} else {
+				$result = $next;
+			}
+
+			@unlink($file);
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Splits a large file up into many smaller files.
+	 *
+	 * @param string $path Path to large file to split
+	 * @param int $lines Number of lines per file
+	 *
+	 * @return array List of file paths
+	 */
+	protected function splitFile($path, $lines = null) {
+		ini_set('auto_detect_line_endings', true);
+
+		if (!is_int($lines)) {
+			$lines = $this->config()->get("lines");
+		}
+
+		$new = $this->getNewSplitFileName();
```

Comment:
I'd feel better about this if the temporary files were cleaned up after processing. It's a fairly generic utility, and hence could be used to upload quite sensitive files. If you look at `getTempParentFolder()`, you'll see that those temporary files can end up in the webroot (`silverstripe-cache`). So while the uniqid makes them fairly hard to guess, it adds unnecessary security surface IMHO.